### PR TITLE
Add a service to trigger functionallity

### PR DESCRIPTION
**Pull request by liamhan0905**
# 🎉 New feature

## Summary

Extended the [trigger publisher](https://github.com/gazebosim/gz-sim/tree/ign-gazebo6/src/systems/triggered_publisher) functionality to allow calling a service. For this example, I've reset the `pose` of the robot as well as set the `cmd_vel` value to `0`, hence stopping the robot. This will allow others to use triggered publisher for calling arbitrary number of services as an "output" from the plugin.

https://user-images.githubusercontent.com/44885838/180519901-f5d5213d-d0ff-4a6d-95f4-19b6358d2d82.mp4

## Test it
I have modified the `tutorial/triggered_publisher.md` tutorial to add this functionality. The world file, `examples/words/triggered_publisher.sdf` also reflects this.

- run `ign topic -t "/start" -m ignition.msgs.Empty -p " "` to run the original triggered publisher example. 
- run `ign topic -t "/reset_robot" -m ignition.msgs.Empty -p " "` to reset the robot (calling the reset pose service). 
  - metadata related to this plugin can be found in the world file. E.g. modifying the reset position

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/systems/triggered_publisher/TriggeredPublisher.cc [hunk 1]
```diff
@@ -596,33 +598,81 @@ void TriggeredPublisher::Configure(const Entity &,
...
+       if (s_info.reqMsg.empty())
+       {
+         ignerr << "Service request string cannot be empty\n";
+       }
+       std::string _timeout = serviceElem->Get<std::string>("timeout");
+       if (_timeout.empty())
+         {
+           ignwarn << "timeout value not spcified for service name ["
+                  << s_info.servName << "] with service type ["
+                  << s_info.reqType << "]. Using default value of 3000\n";
```
Comment (ahcorde) [inline src/systems/triggered_publisher/TriggeredPublisher.cc:634]:
```suggestion
           ignwarn << "Timeout value not specified for service name ["
                   << s_info.servName << "] with service type ["
                   << s_info.reqType << "]. Using default value of 3000\n";
```

File: src/systems/triggered_publisher/TriggeredPublisher.hh [hunk 2]
```diff
@@ -210,31 +223,68 @@ namespace systems
...
+      /// \brief Serivce request type
+      std::string reqType;
+
+      /// \brief Serivce response type
+      std::string repType;
+
+      /// \brief Serivce request message
+      std::string reqMsg;
+
+      /// \brief Service reques timeoutt
```
Comment (ahcorde) [inline src/systems/triggered_publisher/TriggeredPublisher.hh:241]:
```suggestion
      /// \brief Service request type
      std::string reqType;

      /// \brief Service response type
      std::string repType;

      /// \brief Service request message
      std::string reqMsg;

      /// \brief Service request timeout
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
can we add a test ?

**Comment by liamhan0905** [comment]
Yes! Good call. Let me look into adding a test. Thanks!

**Comment by liamhan0905** [comment]
Closing this PR because I managed to mess it up while trying to fix the DCO. Files changed contains commits beyond the scope of this PR. Here's the new PR

**Pull request closed without merging**